specification/_global/knn_search/KnnSearchRequest.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RequestBase } from '@_types/Base'
+import { Fields, Indices, Routing } from '@_types/common'
+import { KnnQuery } from './_types/Knn'
+
+/**
+ * @rest_spec_name knn_search
+ * @since 7.16.0
+ * @stability experimental
+ */
+export interface Request extends RequestBase {
+  path_parts: {
+    /**
+     * A comma-separated list of index names to search;
+     * use `_all` or empty string to perform the operation on all indices
+     */
+    index: Indices
```

Comment:
I guess this should be optional?
```suggestion
    index?: Indices
```

Reply:
This is a mistake in the REST spec description -- I'll make a note to correct it. Only `/<index-pattern>/_knn_search` is supported.